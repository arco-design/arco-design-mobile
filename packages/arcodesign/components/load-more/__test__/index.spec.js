import React from 'react';
import { render, screen,fireEvent } from "@testing-library/react";
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'
import demoTest from '../../../tests/demoTest';
import mountTest from '../../../tests/mountTest';
import LoadMore from '..';

demoTest('load-more');

mountTest(LoadMore, 'LoadMore');

describe('LoadMore type', () => {

    it('Loading state is rendered correctly', () => {
        render(<LoadMore status = 'loading' />)
        const element = screen.getByText('正在努力加载中...')
        expect(element).toBeInTheDocument();
    });

    it('Prepare state is rendered correctly', () => {
        render(<LoadMore status = 'prepare' />)
        const element = screen.getByText('上拉加载更多')
        expect(element).toBeInTheDocument();
    });

    it('Nomore state is rendered correctly', () => {
        render(<LoadMore status = 'nomore' />)
        const element = screen.getByText('没有更多数据了')
        expect(element).toBeInTheDocument();
    });
    
    it('Retry state is rendered correctly', () => {
        render(<LoadMore status = 'retry' />)
        const element = screen.getByText('加载失败，点击重试')
        expect(element).toBeInTheDocument();
    });

    it('BeforeReady state is rendered correctly', () => {
        render(<LoadMore status = 'before-ready' beforeReadyArea = 'before-ready'/>)
        const element = screen.getByText('before-ready')
        expect(element).toBeInTheDocument();
    });

    it('Default state is rendered correctly', () => {
        render(<LoadMore status = 'default' />)
        const element = screen.getAllByRole('generic')
        // 查询的第二个才是正确结果，所以是1
        expect(element[1]).toBeEmptyDOMElement();
    });
})

describe('LoadMore action', () => {

    it('onClick listener correctly', () => {
        const onClick = jest.fn();
        render(<LoadMore onClick = {onClick} throttle = {200} trigger = 'click'/>)
        userEvent.click(screen.getAllByRole('generic')[1])
        expect(onClick).toBeCalled();
        expect(onClick).toBeCalledTimes(1);
    });

    it('onScroll listener correctly',()=>{
        const onChange = jest.fn();
        const { rerender } = render(<LoadMore status='ready' trigger='scroll' onStatusChange={onChange} threshold={0}/>)
        rerender(<LoadMore status = 'prepare' trigger='scroll' onStatusChange={onChange} threshold={0}/>)
        fireEvent.scroll(screen.getAllByRole('generic')[1])
        expect(onChange).toBeCalled();
        expect(onChange).toBeCalledTimes(1);
    })
})