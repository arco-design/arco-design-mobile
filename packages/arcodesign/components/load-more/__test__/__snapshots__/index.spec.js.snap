// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load-more demo test load-more demo: index.md renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="arco-cell-group all-border-box"
    >
      <div
        class="cell-group-body bordered"
      >
        <div
          class="arco-cell all-border-box bordered"
        >
          <div
            class="arco-cell-inner"
          >
            <div
              class="cell-label"
            >
              <div
                class="cell-title"
              >
                Content
              </div>
            </div>
            <div
              class="cell-content has-label"
            />
          </div>
        </div>
        <div
          class="arco-cell all-border-box bordered"
        >
          <div
            class="arco-cell-inner"
          >
            <div
              class="cell-label"
            >
              <div
                class="cell-title"
              >
                Content
              </div>
            </div>
            <div
              class="cell-content has-label"
            />
          </div>
        </div>
        <div
          class="arco-cell all-border-box bordered"
        >
          <div
            class="arco-cell-inner"
          >
            <div
              class="cell-label"
            >
              <div
                class="cell-title"
              >
                Content
              </div>
            </div>
            <div
              class="cell-content has-label"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="arco-load-more status-loading"
      style="padding-top: 16px; padding-bottom: 44px;"
    >
      <div
        class="load-more-text loading"
      >
        正在努力加载中...
      </div>
    </div>
  </div>
</DocumentFragment>
`;
